server/src/main/java/io/crate/action/sql/RowConsumerToResultReceiver.java

```diff
@@ -125,7 +125,9 @@ public void closeAndFinishIfSuspended() {
         if (activeIt != null) {
             activeIt.close();
             completionFuture.complete(null);
-            resultReceiver.allFinished();
+            // resultReceiver is left untouched:
+            // - A previous .batchCompleted() call already flushed out pending messages
+            // - Calling failure/allFinished would lead to extra messages to the client
```

Comment:
just a suggestion: 
```suggestion
            // - Calling failure/allFinished would lead to extra messages, including  sentCommandComplete, to the client, which can lead to issues on the client.
```